include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -51,37 +62,45 @@ template <typename Upstream>
 class pool_memory_resource final
   : public detail::stream_ordered_memory_resource<pool_memory_resource<Upstream>,
                                                   detail::coalescing_free_list> {
- public:
-  friend class detail::stream_ordered_memory_resource<pool_memory_resource<Upstream>,
-                                                      detail::coalescing_free_list>;
-
   static constexpr size_t default_initial_size = ~0;
   static constexpr size_t default_maximum_size = ~0;
   // TODO use rmm-level def of this.
   static constexpr size_t allocation_alignment = 256;
 
+ public:
+  friend class detail::stream_ordered_memory_resource<pool_memory_resource<Upstream>,
+                                                      detail::coalescing_free_list>;
+
   /**
-   * @brief Construct a `pool_memory_resource` and allocate the initial
-   * device memory pool using `upstream_mr`.
+   * @brief Construct a `pool_memory_resource` and allocate the initial device memory pool using
+   * `upstream_mr`.
    *
    * @throws rmm::logic_error if `upstream_mr == nullptr`
    *
    * @param upstream_mr The memory_resource from which to allocate blocks for the pool.
-   * @param initial_pool_size Size, in bytes, of the initial pool. When
-   * zero, an implementation-defined pool size is used.
-   * @param maximum_pool_size Maximum size, in bytes, that the pool can grow to.
+   * @param initial_pool_size Minimum size, in bytes, of the initial pool. Defaults to an
+   * implementation defined size.
```

Comment:
Why not document that the initial size is 1/2 total device memory and max size is total device memory? I don't remember when this "implementation defined size" doc slipped in.

Reply:
I was just preserving what was there. I'm happy to update it.